kernel-api/src/main/scala/org/apache/toree/interpreter/broker/BrokerState.scala

```diff
@@ -49,7 +52,7 @@ class BrokerState(private val maxQueuedCode: Int) {
    *
    * @return The future containing the results of the execution
    */
-  def pushCode(code: Code): Future[CodeResults] = synchronized {
+  def pushCode(code: Code, outputResultStream: Option[OutputStream]): Future[CodeResults] = synchronized {
```

Comment:
Rather than requiring None as a second argument everywhere, can you have this be an overloaded method where one instance has a signature of `def pushCode(code: Code)` with the output stream set to None or just have the `outputResultStream: Option[OutputStream]` be set to a default value of None in this function? Don't want to have to put None in explicitly everywhere.